From 99e2a5e0f4adfa3f3027b4691eaefa05e3d1fab2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Thu, 29 Dec 2022 00:47:28 +0100
Subject: [PATCH 5/8] gio: fix duplicate g_trace_* symbols in static libs with
 --default-library=both

Should fix ld.gold linker errors linking static glib/gio into an app:

error: /builds/cerbero-android-universal/arm64/lib/libglib-2.0.a(gtrace.c.o): multiple definition of 'g_trace_mark'
/builds/cerbero-android-universal/arm64/lib/libgio-2.0.a(.._glib_gtrace.c.o): previous definition here

error: /builds/cerbero-android-universal/arm64/lib/libglib-2.0.a(gtrace.c.o): multiple definition of 'g_trace_define_int64_counter'
/builds/cerbero-android-universal/arm64/lib/libgio-2.0.a(.._glib_gtrace.c.o): previous definition here

error: /builds/cerbero-android-universal/arm64/lib/libglib-2.0.a(gtrace.c.o): multiple definition of 'g_trace_set_int64_counter'
/builds/cerbero-android-universal/arm64/lib/libgio-2.0.a(.._glib_gtrace.c.o): previous definition here
---
 gio/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/meson.build b/gio/meson.build
index 1c9a369..e8b7a16 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -610,7 +610,7 @@ gio_base_sources = files(
 
 gio_sources = gio_base_sources
 
-if glib_build_shared
+if not glib_build_static
   gio_sources += files ('../glib/gtrace.c')
 endif
 
-- 
2.52.0

